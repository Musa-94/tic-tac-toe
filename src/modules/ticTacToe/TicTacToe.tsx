import React from 'react';
import { withLocalization } from '../../lang/localization';
import { TGameBoard } from '../../types';
import {
    Row,
    Symbol,
    GameBoard,
} from './styledComponents';

type TProps = {
    currentGameBoard: TGameBoard
}

const TicTacToe: React.FC<TProps> = props => {
    const {
        currentGameBoard
    } = props;

    return (
        <GameBoard>
            <Row>
                {
                    currentGameBoard[0].map((symbol, index) =>
                        <Symbol key={index}
                                children={symbol === '-' ? '' : symbol}
                        />
                    )
                }
            </Row>
            <Row>
            {
                currentGameBoard[1].map((symbol, index) =>
                    <Symbol key={index}
                            children={symbol === '-' ? '' : symbol}
                    />
                )
            }
            </Row>
            <Row>
            {
                currentGameBoard[2].map((symbol, index) =>
                    <Symbol key={index}
                            children={symbol === '-' ? '' : symbol}
                    />
                )
            }
            </Row>
        </GameBoard>
    )
}

export default React.memo(withLocalization()(TicTacToe));
