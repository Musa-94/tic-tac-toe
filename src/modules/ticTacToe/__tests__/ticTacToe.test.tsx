import React from 'react';
import TicTacToe from '../TicTacToe';

const props = {
    isWin: false,
    setPlayerSymbol: () => {},
    currentGameBoard: {
        0: ['', '', ''],
        1: ['', '', ''],
        2: ['', '', ''],
    },
    currentPlayerSymbol: 'x',
};

describe('TicTacToe snapshot', () => {
    console.log = jest.fn();

    it('should correctly render', () => {
        //@ts-ignore TODO remove when learn write to declare
        const component = renderer(<TicTacToe {...props}/>);

        expect(component).toMatchSnapshot();
    });
});
