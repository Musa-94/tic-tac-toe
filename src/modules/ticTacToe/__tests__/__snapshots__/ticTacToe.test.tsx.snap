// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicTacToe snapshot should correctly render 1`] = `
.c0 {
  width: 100%;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  width: 100%;
  height: 33.3%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  width: 100%;
  height: 33.3%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-top: 2px solid black;
  border-bottom: 2px solid black;
}

.c2 {
  width: 33.3%;
  height: 100%;
  font-size: 10em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFC600;
}

.c2:hover {
  opacity: 0.6;
}

.c3 {
  width: 33.3%;
  height: 100%;
  font-size: 10em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFC600;
  border-left: 2px solid black;
  border-right: 2px solid black;
}

.c3:hover {
  opacity: 0.6;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="c2"
      onClick={[Function]}
    >
      
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="c2"
      onClick={[Function]}
    >
      
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      
    </div>
    <div
      className="c2"
      onClick={[Function]}
    >
      
    </div>
  </div>
</div>
`;
